<launch>
  <param name="/use_sim_time" value="true" />

  <param name="/robot_description" textfile="$(find planning_models)/test_urdf/robot.xml" />

  <!-- send parameters for multidof -->
  <rosparam command="load" ns="robot_description_planning" file="$(find pr2_arm_navigation_config)/config/pr2_planning_description.yaml" />

  <rosparam command="load" file="$(find pr2_arm_navigation_config)/config/joint_limits.yaml"/>

  <node pkg="planning_environment" name="wall_clock_server" type="fake_time.py" />

  <node pkg="planning_environment" type="environment_server" output="screen" name="environment_server">      
    <param name="use_monitor" type="bool" value="false"/>
  </node>

  <node pkg="planning_environment" type="planning_scene_validity_server" output="screen" name="planning_scene_validity_server"/>      
  
  <node pkg="pr2_arm_kinematics_constraint_aware" type="pr2_arm_kinematics_constraint_aware" name="pr2_right_arm_kinematics" output="screen">    
   	 <!-- IK parameters -->
   	 <param name="tip_name" value="r_wrist_roll_link" />     
   	 <param name="root_name" value="torso_lift_link" />  

	 <param name="group" type="string" value="right_arm" />
   </node>
   <node pkg="pr2_arm_kinematics_constraint_aware" type="pr2_arm_kinematics_constraint_aware" name="pr2_left_arm_kinematics" output="screen">    
   	 <!-- IK parameters -->
   	 <param name="tip_name" value="l_wrist_roll_link" />     
   	 <param name="root_name" value="torso_lift_link" />  

	 <param name="group" type="string" value="left_arm" />
   </node>
   <node pkg="ompl_ros_interface" type="ompl_ros" name="ompl_planning" output="screen">
     <rosparam command="load" file="$(find pr2_arm_navigation_planning)/config/ompl_planning.yaml" />
   </node>
  <node name="interpolated_ik_node_right" pkg="interpolated_ik_motion_planner" 
    type="interpolated_ik_motion_planner.py" args="r" respawn="false"/>
   <node pkg="trajectory_filter_server" name="trajectory_filter" type="trajectory_filter_server" output="screen">
     <rosparam command="load" file="$(find pr2_arm_navigation_config)/config/joint_limits.yaml"/>
     <rosparam command="load" file="$(find pr2_arm_navigation_filtering)/config/filters.yaml"/>
   </node>
  <node name="collision_proximity_server_test" pkg="collision_proximity" type="collision_proximity_server_test" output="screen" respawn="false" />
  <node name="collision_proximity_planner" pkg="collision_proximity_planner" type="collision_proximity_planner" output="screen" respawn="false" />

  <node name="mongo" type="wrapper.py" pkg="mongodb">
    <param name="overwrite" value="false"/>
  </node>

  <param name="warehouse_port" value="27019"/>

  <node pkg="robot_state_publisher" type="state_publisher" name="rob_st_pub" />

</launch>
